### ISSUE TITLE
Record/replay segfaults when built with --disable-tracing

### ISSUE THREAD

**Issue by stwhite91** [UIUC-PPL/charm#2446]
```
   frame #0: 0x00000001000ef491 pgm`openReplayFile(prefix="ckreplay_", suffix=".log", permissions="w") at ck.C:2543:23
   2540
   2541 #if CMK_REPLAYSYSTEM
   2542 static FILE *openReplayFile(const char *prefix, const char *suffix, const char *permissions) {
-> 2543   std::string fName = CkpvAccess(traceRoot);
   2544   fName += prefix;
   2545   fName += std::to_string(CkMyPe());
   2546   fName += suffix;
  thread #3, stop reason = EXC_BAD_ACCESS (code=1, address=0x10)
    frame #0: 0x00000001000ef491 pgm`openReplayFile(prefix="ckreplay_", suffix=".log", permissions="w") at ck.C:2543:23
   2540
   2541 #if CMK_REPLAYSYSTEM
   2542 static FILE *openReplayFile(const char *prefix, const char *suffix, const char *permissions) {
-> 2543   std::string fName = CkpvAccess(traceRoot);
   2544   fName += prefix;
   2545   fName += std::to_string(CkMyPe());
   2546   fName += suffix;
```

**Comment by ericjbohm** [comment]
Record replay depends on parts of the tracing framework in a variety of places. So I think the right answer here is to reject attempts to compile charm that way.

**Comment by matthiasdiener** [comment]
Fixed via #2449.
